### PR TITLE
chore: implement state selection if US country selected

### PR DESCRIPTION
**Pull request by dangelovska**

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressViewController.swift [hunk 1]
```diff
@@ -91,6 +91,8 @@ class KYCAddressViewController: BaseTableViewController<KYCCoordinator,
             view.configure(with: .init())
             view.setup(with: model)
             
+            view.secondaryTextField.isUserInteractionEnabled = dataStore?.country != C.countryUS
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressViewController.swift:94]: please handle this in setup

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressStore.swift [hunk 2]
```diff
@@ -25,6 +26,7 @@ class KYCAddressStore: NSObject, BaseDataStore, KYCAddressDataStore {
     var country: String?
     var countryFullName: String?
     var ssn: String?
+    var stateName: String?
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressStore.swift:29]: move this after line 24and specify which is a code which is a full name

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift [hunk 3]
```diff
@@ -68,9 +68,9 @@ class KYCAddressInteractor: NSObject, Interactor, KYCAddressViewActions {
                                           address: dataStore?.address ?? "",
                                           city: dataStore?.city ?? "",
                                           zip: dataStore?.postalCode ?? "",
-                                          country: dataStore?.country ?? "",
+                                          country: dataStore?.countryFullName ?? "",
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift:71]: are you sure?

File: brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/Cycles/CountriesAndStates/CountriesAndStatesVIP.swift [hunk 4]
```diff
@@ -12,24 +12,30 @@ import Foundation
...
     func displayCountry(responseDisplay: CountriesAndStatesModels.SelectCountry.ResponseDisplay)
+    func displayState(responseDisplay: CountriesAndStatesModels.SelectState.ResponseDisplay)
 }
 
 protocol CountriesAndStatesDataStore: FetchDataStore {
     var country: String? { get set }
     var countryFullName: String? { get set }
     var countries: [Country] { get set }
+    var states: [USState] { get set }
+    var stateName: String? { get set }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/Cycles/CountriesAndStates/CountriesAndStatesVIP.swift:33]: we never need the code?

File: brd-ios/breadwallet/src/Redesign/Views/DoubleHorizontalTextboxView.swift [hunk 5]
```diff
@@ -63,7 +63,7 @@ class DoubleHorizontalTextboxView: FEView<DoubleHorizontalTextboxViewConfigurati
         return view
     }()
     
-    private lazy var secondaryTextField: FETextField = {
+    lazy var secondaryTextField: FETextField = {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/DoubleHorizontalTextboxView.swift:66]: privatize this please

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift [hunk 6]
```diff
@@ -62,15 +59,16 @@ class KYCAddressInteractor: NSObject, Interactor, KYCAddressViewActions {
     }
     
     func submitInfo(viewAction: KYCAddressModels.Submit.ViewAction) {
+        let state = dataStore?.country == C.countryUS ? dataStore?.stateCode : dataStore?.state
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift:62]: why?

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressViewController.swift [hunk 7]
```diff
@@ -83,12 +83,13 @@ class KYCAddressViewController: BaseTableViewController<KYCCoordinator,
...
         let section = sections[indexPath.section]
         guard let cell: WrapperTableViewCell<DoubleHorizontalTextboxView> = tableView.dequeueReusableCell(for: indexPath),
-              let model = sectionRows[section]?[indexPath.row] as? DoubleHorizontalTextboxViewModel else {
+              var model = sectionRows[section]?[indexPath.row] as? DoubleHorizontalTextboxViewModel else {
             return UITableViewCell()
         }
         
         cell.setup { view in
             view.configure(with: .init())
+            model.isUserInteractionEnabled = dataStore?.country != C.countryUS
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressViewController.swift:92]: please, do all that in setup somehow

File: brd-ios/breadwallet/src/Redesign/Views/DoubleHorizontalTextboxView.swift [hunk 8]
```diff
@@ -117,6 +118,10 @@ class DoubleHorizontalTextboxView: FEView<DoubleHorizontalTextboxViewConfigurati
         primaryTextField.setup(with: viewModel?.primary)
         secondaryTextField.setup(with: viewModel?.secondary)
         
+        if let isUserInteractionEnabled = viewModel?.isUserInteractionEnabled {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/DoubleHorizontalTextboxView.swift:121]: this logic does not say much. please, update it with more descriptive code.

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift (hunk 6 above)
Comment (dangelovska) [reply brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift:62]: For US states we send the code from the state and for the others the name.

**Review by kenanmamedoff — DISMISSED** [review]

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**